lib/main_common.pm

```diff
@@ -3221,4 +3222,18 @@ sub load_kernel_baremetal_tests {
     loadtest "kernel/build_git_kernel" if get_var('KERNEL_GIT_TREE');
 }
 
+sub setup_yui_rest_api {
+    if (check_var('BACKEND', 'qemu')) {
+        # On qemu we connect to the worker using port forwarding
+        set_var('YUI_SERVER', 'localhost');
+        my $yuiport = get_var('YUI_START_PORT', 39000);
+        $yuiport += get_var('VNC') =~ /(?<vncport>\d+)/ ? $+{vncport} : int(rand(1000));
+        die "Cannot set port for YUI REST API" unless $yuiport;
+        set_var('YUI_PORT', $yuiport);
+        set_var('EXTRABOOTPARAMS', get_var('EXTRABOOTPARAMS', '')
+              . " startshell=1 YUI_HTTP_PORT=$yuiport YUI_HTTP_REMOTE=1 YUI_REUSE_PORT=1");
```

Comment:
just thinking: would this be more suitable in `get_extra_boot_params` of bootloader_setup.pm?

Reply:
`get_extra_boot_params` reads the value and returns it, here we set it, so there is no good reason to mix those two things.

Reply:
i guess it does not make much difference at the moment. to justify my suggestion, i just wanted your opinion as i believe there it could a place to have all the extra boot params aggregated(if you know what i mean). `get_bootmenu_console_params` for instance is also a *getter* in the same sense but it is used to adjust the @params and return it.